src/wallet/scriptpubkeyman.cpp

```diff
@@ -290,20 +293,72 @@ bool LegacyScriptPubKeyMan::GetReservedDestination(const OutputType type, bool i
     return true;
 }
 
+bool LegacyScriptPubKeyMan::TopUpInactiveHDChain(const CKeyID seed_id, int64_t index, bool internal)
+{
+    LOCK(cs_KeyStore);
```

Comment:
This is only called from within `MarkUnusedAddresses()`, which has already taken this lock. Rather than taking the lock recursively, I think it makes more sense to add an `EXCLUSIVE_LOCKS_REQUIRED` annotation.